src/test/java/com/puppycrawl/tools/checkstyle/XpathRegressionTest.java

```diff
@@ -0,0 +1,199 @@
+            bw.write("       checks=\"");
+            bw.write(checkName);
+            bw.write("\"\n");
+            bw.write("       query=\"");
+            bw.write(xpathQueries.stream().collect(Collectors.joining(DELIMITER)));
+            bw.write("\"/>\n");
+            bw.write("</suppressions>");
+        }
+
+        return suppressionsXpathConfigFile.getPath();
+    }
+
+    private void verifyViolation(DefaultConfiguration checkerConfig, File file,
+                                 String... expectedViolation)
+            throws Exception {
+
+        verify(checkerConfig,
+                file.getPath(),
+                expectedViolation);
+    }
+
+    private DefaultConfiguration createSuppressionXpathFilter(String checkName,
+                                           List<String> xpathQueries) throws Exception {
+        final DefaultConfiguration suppressionXpathFilterConfig =
+                createModuleConfig(SuppressionXpathFilter.class);
+
+        suppressionXpathFilterConfig.addAttribute("file",
+                createSuppressionsXpathConfigFile(checkName, xpathQueries));
+
+        return suppressionXpathFilterConfig;
+    }
+
+    private void runVerifications(DefaultConfiguration moduleConfig,
+                                  CheckVariables variables,
+                                  String[] expectedViolation,
+                                  List<String> expectedXpathQueries) throws Exception {
+        final DefaultConfiguration checkerConfig =
+                createCheckerWithTreeWalkerModules(moduleConfig);
+
+        final List<String> generatedXpathQueries =
+                generateXpathQueries(variables.fileToProcess, variables.violationLineNumber,
+                        variables.violationColumnNumber);
+
+        final DefaultConfiguration checkerConfigWithXpath =
+                createCheckerWithTreeWalkerModules(moduleConfig,
+                        createSuppressionXpathFilter(variables.checkName, generatedXpathQueries));
+
+        verifyViolation(checkerConfig, variables.fileToProcess, expectedViolation);
+        verifyXpathQueries(generatedXpathQueries, expectedXpathQueries);
+        verifyViolation(checkerConfigWithXpath, variables.fileToProcess, NO_VIOLATIONS);
+    }
+
+    @Test
+    public void testCyclomaticComplexityCheck() throws Exception {
+        final String checkName = CyclomaticComplexityCheck.class.getName();
+        final File fileToProcess =
+                new File(getPath("InputSuppressionXpathFilterCyclomatic.java"));
+        final int violationLineNumber = 4;
+        final int violationColumnNumber = 5;
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(CyclomaticComplexityCheck.class);
+        moduleConfig.addAttribute("max", "0");
+
+        final String[] expectedViolation = {
+            "4:5: " + getCheckMessage(CyclomaticComplexityCheck.class,
+                    CyclomaticComplexityCheck.MSG_KEY, 2, 0),
```

Comment:
```
+        final int violationLineNumber = 4;
+        final int violationColumnNumber = 5;
..
+            "4:5: " + getCheckMessage
```

unfortunate copy-paste, can we eliminate it ?

Why you limit design only to one violation in file ?